android/KMAPro/kMAPro/src/main/java/com/tavultesoft/kmapro/KeymanSettingsFragment.java

```diff
@@ -87,6 +128,14 @@ String getInstalledLanguagesText() {
   public void onResume() {
     super.onResume();
 
+    update();
+  }
+
+  public void update() {
+    setSystemKeyboardPreference.setChecked(GetStartedActivity.isEnabledAsSystemKB(context));
```

Comment:
I think it would be good if the Settings activity didn't refer to the GetStarted activity but rather we had a separate class that implemented the `isEnabledAsSystemKB` and `isDefaultKB` functions? It's not that healthy to have cross-references like that :)

Reply:
... have you **seen** KMEA's keyboard handling code?  _shudder_

Granted, yeah, it would be nice to not introduce more of that.  So, just a small, isolated class with those two methods?  Or should I throw them into `MainActivity`, since that's the app's KMManager equivalent?

Reply:
> ... have you **seen** KMEA's keyboard handling code? _shudder_

Indeed I have :) And we have a long way to go to bring that to a place of beauty ... as you say, let's not make it worse. I don't mind all that much whether you plug them into `MainActivity` or into a small isolated class; both are cleaner than the current. Long term a small isolated class is cleaner and more mockable.